Makefile.base

```diff
@@ -70,7 +70,7 @@ $(BINDIR)/$(MODULE).a $(OBJ): | $(BINDIR)/$(MODULE)/
 $(BINDIR)/$(MODULE).a: $(OBJ) | $(DIRS:%=ALL--%)
 	@# Recreate archive to cleanup deleted/non selected source files objects
 	$(Q)$(RM) $@
-	$(Q)$(AR) $(ARFLAGS) $@ $^
+	$(Q)$(AR) $(ARFLAGS) $(foreach f,$(abspath $@ $^),$(call relpath,$(abspath .),$f))
```

Comment:
The `relpath` implementation should take care of the `foreach` like most makefile functions do.
Also I would prefer the python arguments ordering https://docs.python.org/3.7/library/os.path.html#os.path.relpath

```
relpath, directories, optional_ref_directory
```
Where optional_ref_directory is replaced by `$(abspath .)` if it is empty. It can be done with `$(if $2,$2,$(abspath .))` in the function call.